examples/foundational/07b-interruptible-langchain.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright (c) 2024, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+
+import asyncio
+import os
+import sys
+
+import aiohttp
+from dotenv import load_dotenv
+from loguru import logger
+from runner import configure
+
+from pipecat.frames.frames import LLMMessagesFrame
+from pipecat.pipeline.pipeline import Pipeline
+from pipecat.pipeline.runner import PipelineRunner
+from pipecat.pipeline.task import PipelineTask
+from pipecat.processors.aggregators.llm_response import (
+    LLMAssistantResponseAggregator, LLMUserResponseAggregator)
+from pipecat.services.elevenlabs import ElevenLabsTTSService
+from pipecat.services.langchain import LangchainProcessor
+from pipecat.transports.services.daily import DailyParams, DailyTransport
+from pipecat.vad.silero import SileroVADAnalyzer
+
+load_dotenv(override=True)
+
+try:
+    from langchain.prompts import ChatPromptTemplate, MessagesPlaceholder
+    from langchain_community.chat_message_histories import ChatMessageHistory
+    from langchain_core.chat_history import BaseChatMessageHistory
+    from langchain_core.runnables.history import RunnableWithMessageHistory
+    from langchain_openai import ChatOpenAI
+
+except ModuleNotFoundError as e:
+    logger.exception(
+        "You need to `pip install langchain langchain-openai langchain-community` for this example. Also, be sure to set `OPENAI_API_KEY` in the environment variable."
```

Comment:
I believe this could just be `pip install pipecat-ai[langchain]` since you already add the dependency in pyproject.yml

Reply:
Oh, now that I see i this try/catch should actually go in `src/pipecat/services/langchain.py` as other services. Not necessary to add it here if you don't want to. In theory, if you follow the isntructions this should be installed for you.

I'll update the requirements files for both macos and linux.

Reply:
I did add the try/catch in the service file, however just testing for `langchain`. The things from langchain-openai and langchain-community used in the example are not mandatory and imo should hence not be part of the core requirements. Therefor, I removed langchain-community from the pyptoject.yaml again. It's not needed by the service, but only/probably/maybe by user code, in this case to keep track of the message history.
I have no clue why LC decided to put `ChatMessageHistory` in the community package …